pygments/lexers/dotnet.py

```diff
@@ -704,3 +704,29 @@ def analyse_text(text):
             result += 0.05
 
         return result
+
+
+class BlazorLexer(RegexLexer):
+    """
+    For `Blazor <https://docs.microsoft.com/en-us/aspnet/core/blazor/?view=aspnetcore-5.0>`_
+    source code.
+    """
+
+    name = 'Blazor'
+    aliases = ['blazor']
+    filenames = ['*.razor']
+    mimetypes = []
+
+    tokens = {
+        'root': [
+            (r'(@code |@page )', Keyword),
+            (r'.+', using(CSharpLexer)),
```

Comment:
This only matches whole lines so it likely won't be able to parse all kinds of C#. I guess you need to push and pop braces to lex the inner content as C# and the rest as HTML.